### DISCUSSION TITLE
Add Controls/Key Mapping Support for GenyMotion Desktop

### DISCUSSION THREAD

**Discussion by alizaincodes** [Genymobile/feedback#158] [Genymotion Desktop] [answered]
Hello :wave: ,
I have seen many other discussions related to controls and key mapping.
But no one answered there. So I am here to ask you. 
Can you add support for Controls or Key Mapping in GenyMotion Desktop just like in other emulators like Bluestacks?
I have seen many gamers on youtube saying that genymotion is very very lightweight emulator so you can use it to play games.
But there is a big problem for controls and key mapping. 

Key Mapping means to display letters on the screen and when that key is pressed through keyboard, tap on that specific location where the letter is displayed.
It is just like BlueStacks.
See the image below:
![image](https://github.com/Genymobile/feedback/assets/139030625/2ecd7cfc-02ea-4bbd-9ede-6da22b82c93f)

See there are small tiny letters are placed beside buttons so can user press the key on keyboard and tap on that specific place.

There is also an option for mouse that whenever the mouse moves the character also rotates or changes its direction in the game.
When mouse is clicked, the key binded or key mapped on the screen such as fire button clicks.

If you can't add this feature, you should add a feature to insert the keyboard and mouse into the emulator so can apps like octopus keymapper can do their job.

Please add this feature so the Gamers can also use GenyMotion.
Thanks!

**Comment by agavignet** [comment, marked as answer]
Hi @alizaincodes 

Thank you for your detailed feedback. We're pleased to see your interest in this feature. I want to let you know that it is on our roadmap, but due to its current priority, I can't provide an ETA. However, when the time comes, you'll be informed, and rest assured that your advice will be considered in the feature's design.

Adrien
